optimum/intel/openvino/modeling_base.py

```diff
@@ -186,6 +193,42 @@ def fix_op_names_duplicates(model: openvino.runtime.Model):
 
         return model
 
+    @staticmethod
+    def _compile_model(
+        file_name: Union[str, Path],
+        device: Optional[str] = None,
+        ov_config: Optional[Dict[str, str]] = None,
+        allow_set_cache_dir=False,
+        model_save_dir: Union[str, Path] = None,
+    ):
+        logger.info(f"Compiling the model to {device} ...")
+        if isinstance(file_name, str):
+            file_name = Path(file_name)
+        ov_config = ov_config or {}
+
+        if model_save_dir is None:
+            model_save_dir = file_name.parent
+        if "CACHE_DIR" not in ov_config.keys() and (
+            allow_set_cache_dir
+            or (not str(model_save_dir).startswith(gettempdir()) and (device is not None and "gpu" in device.lower()))
+        ):
+            # Set default CACHE_DIR only if it is not set, if the model is not in a temporary directory, and device is GPU
+            cache_dir = Path(model_save_dir).joinpath("model_cache")
+            ov_config["CACHE_DIR"] = str(cache_dir)
+            logger.info(f"Setting OpenVINO CACHE_DIR to {str(cache_dir)}")
+        if file_name.suffix == ".onnx":
+            compiled_model = core.compile_model(
+                file_name, device.upper() if device is not None else device, config=ov_config
+            )
+        else:
+            compiled_model = core.compile_model(
+                file_name, device.upper() if device is not None else device, config=ov_config
+            )
```

Comment:
why the condition if it's the same behavior ?

Reply:
removed